tests/rbd/test_rbd_image_operations.py

```diff
@@ -0,0 +1,295 @@
+            ops_module="ceph.rbd.workflows.rbd",
+            ops_method="run_io_and_check_rbd_status",
+            test_ops_parallely=test_ops_parallely,
+        )
+        if rc:
+            log.error(f"Run IOs and verify rbd status failed for pool {pool}")
+            return 1
+
+        log.info(f"Remove images and verify for pool {pool}")
+        rc = wrapper_for_image_ops(
+            rbd=rbd,
+            pool=pool,
+            image_config=multi_image_config,
+            client=client,
+            ops_module="ceph.rbd.workflows.rbd",
+            ops_method="remove_single_image_and_verify",
+            test_ops_parallely=test_ops_parallely,
+        )
+        if rc:
+            log.error(f"Remove images and verify rbd status failed for pool {pool}")
+            return 1
+    except Exception as e:
+        log.error(f"Testing image operations failed for pool {pool} with error {e}")
+        if test_ops_parallely:
+            raise Exception(
+                f"Testing image operations failed for pool {pool} with error {e}"
+            )
+        return 1
+
+
+def test_rbd_image_operations(rbd_obj, client, **kw):
+    """
+    Test all image operations on each pool created
+
+    Args:
+        rbd_obj: rbd object
+        client: client node
+        **kw: key word arguments
+    """
+    log.info(f"Performing image operations for pool type {rbd_obj['pool_types']}")
+    rbd = rbd_obj.get("rbd")
+    for pool_type in rbd_obj.get("pool_types"):
+        rbd_config = kw.get("config", {}).get(pool_type, {})
+        multi_pool_config = getdict(rbd_config)
+        test_ops_parallely = rbd_config.get("test_ops_parallely", False)
+        if test_ops_parallely:
+            with parallel() as p:
+                for pool, pool_config in multi_pool_config.items():
+                    p.spawn(
+                        test_rbd_image_operations_for_pool,
+                        rbd=rbd,
+                        pool=pool,
+                        pool_config=pool_config,
+                        test_ops_parallely=test_ops_parallely,
+                        client=client,
+                        pool_type=pool_type,
+                        **kw,
+                    )
+        else:
+            for pool, pool_config in multi_pool_config.items():
+                rc = test_rbd_image_operations_for_pool(
+                    rbd=rbd,
+                    pool=pool,
+                    pool_config=pool_config,
+                    test_ops_parallely=test_ops_parallely,
+                    client=client,
+                    pool_type=pool_type,
+                    **kw,
+                )
+                if rc:
+                    log.error(f"Image operations testing failed for pool {pool}")
+                    return 1
+    return 0
+
+
+def run(**kw):
+    """RBD image operations testing.
+
+    Args:
+        **kw: test data
+    """
+    log.info("Running test <polarion_id> - Testing RBD image operations.")
```

Comment:
please add poarion id?

Reply:
I'm still thinking whether we can create new polarion test case or use any of the existing ones. Once I get clarity, I will add this. Since baremetal pipeline won't run anytime soon, we can merge this for now, later will add polarion id to this. before adding it to pipeline.